### PR TITLE
Add support for deepObject

### PR DESCRIPTION
**Pull request by ctipggroup**
Add support for deepObject style
Add test

### CODE REVIEW

**Comment by ctipggroup** [comment]
Add support for single level deepObject style Query Parameters. e.g. /users?id[role]=admin&id[firstName]=Alex

File: tests/FromCommunity/IssueWithQueryArrayTest.php [hunk 1]
```diff
@@ -12,62 +12,183 @@ final class IssueWithQueryArrayTest extends TestCase
...
         $this->addToAssertionCount(1);
     }
 
     public function testConvertFormIntegerArrayError() : void
     {
         $this->expectExceptionMessage('Value "id1,id2,id3" for argument "id" is invalid for Request [get /users]');
-        $validator = (new ValidatorBuilder())->fromYaml($this->makeYaml('form', 'integer', 'int32'))->getServerRequestValidator();
+        $validator = (new ValidatorBuilder())->fromYaml($this->makeArrayYaml('form', 'integer', 'int32'))->getServerRequestValidator();
         $validator->validate($this->makeRequest('form', 'string'));
         $this->addToAssertionCount(1);
```
Comment (scaytrase) [inline tests/FromCommunity/IssueWithQueryArrayTest.php:53]: Didn't noticed this on prev one, but the same thing is here for new tests. There is no need in this call or `error` test - this line of code is not reachable at all. Could you please clean it up?

File: tests/FromCommunity/IssueWithQueryArrayTest.php [hunk 2]
```diff
@@ -12,62 +12,183 @@ final class IssueWithQueryArrayTest extends TestCase
...
+      responses:
+        '200':
+          description: A list of users
+YAML;
+        $validator = (new ValidatorBuilder())->fromYaml($yaml)->getServerRequestValidator();
+        $validator->validate($this->makeRequest('deepObject', 'integer'));
+        $this->addToAssertionCount(1);
+    }
+
+    public function testConvertMultiLayerDeepObject() : void
```
Comment (scaytrase) [inline tests/FromCommunity/IssueWithQueryArrayTest.php:107]: Can you please add a error test where only the deepest element is not valid

File: tests/FromCommunity/IssueWithQueryArrayTest.php [hunk 3]
```diff
@@ -12,62 +12,183 @@ final class IssueWithQueryArrayTest extends TestCase
...
+                type: integer
+                format: int32
+              after:
+                type: integer
+                format: int32
+      responses:
+        '200':
+          description: A list of users
+YAML;
+        $this->expectExceptionMessage('Value "{
```
Comment (scaytrase) [inline tests/FromCommunity/IssueWithQueryArrayTest.php:182]: I'm not sure whether it's good explanation of what is going wrong with parameter. Can we also inspect prev exceptions to examine correct validations ?

File: tests/FromCommunity/IssueWithQueryArrayTest.php (hunk 1 above)
Comment (ctipggroup) [reply tests/FromCommunity/IssueWithQueryArrayTest.php]: Removed as requested

File: tests/FromCommunity/IssueWithQueryArrayTest.php (hunk 2 above)
Comment (ctipggroup) [reply tests/FromCommunity/IssueWithQueryArrayTest.php:156]: Updated test to break on deepest element

File: tests/FromCommunity/IssueWithQueryArrayTest.php (hunk 3 above)
Comment (ctipggroup) [reply tests/FromCommunity/IssueWithQueryArrayTest.php]: Updated error tests to inspect previous exceptions instead of error message.

**Comment by ctipggroup** [comment]
> Great work! In general looks good but some minor changes needed

Changes made as requested.

Thanks

**Comment by scaytrase** [comment]
Sorry for delay, was a bit busy

**Comment by scaytrase** [comment]
LGTM 👍 
Great work @ctipggroup

**Pull request merged**